tf2_ros/src/static_transform_broadcaster_program.cpp

```diff
@@ -35,68 +35,172 @@
 #include <cstdio>
 #include <memory>
 #include <string>
+#include <unordered_map>
+#include <utility>
 #include <vector>
 
+std::unordered_map<std::string, std::string> _make_arg_map(std::vector<std::string> && args)
+{
+  std::unordered_map<std::string, std::string> ret;
+  /* collect from [exe] --option1 value --option2 value ... --optionN value */
+  for (size_t x = 1; x < args.size(); x += 2) {
+    ret.emplace(std::move(args[x]), std::move(args[x + 1]));
+  }
```

Comment:
What happens here if the user passes an incorrect command line, like:

```
--x --y 0
```

Reply:
Ah, that's a good catch. Would likely crash.